great_expectations/experimental/datasources/spark_azure_blob_storage_datasource.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+import pydantic
+from typing_extensions import Literal
+
+from great_expectations.core.util import AzureUrl
+from great_expectations.experimental.datasources import _SparkFilePathDatasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    AzureBlobStorageDataConnector,
+    DataConnector,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.spark_datasource import (
+    SparkDatasourceError,
+)
+from great_expectations.experimental.datasources.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSorter,
+        BatchSortersDefinition,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+ABS_IMPORTED = False
+try:
+    from azure.storage.blob import BlobServiceClient, ContainerClient
+
+    ABS_IMPORTED = True
+except ImportError:
+    BlobServiceClient = None
+    ContainerClient = None
```

Comment:
You shouldn't need to set these to `None`.
Do them as `TYPE_CHECKING` imports.

Reply:
@Kilo59 Please see https://github.com/great-expectations/great_expectations/pull/7255#discussion_r1121853785 (and `https://youtrack.jetbrains.com/issue/PY-8933/Import-in-try-block-should-also-be-defined-in-except-block-false-positive-when-name-not-used`).  Thank you.